inventory/nodes_test.go

```diff
@@ -1,19 +1,326 @@
+			Context: context.TODO(),
+		})
+		require.NoError(t, err)
+		require.NotNil(t, changeRes.Payload.Container)
+		require.Equal(t, getNodeRes.Payload.Container.NodeID, res.Payload.Container.NodeID)
+		require.Equal(t, getNodeRes.Payload.Container.NodeName, "Changed Container Node")
+	})
+
+	t.Run("AddNameEmpty", func(t *testing.T) {
+		params := &nodes.AddContainerNodeParams{
+			Body:    nodes.AddContainerNodeBody{NodeName: ""},
+			Context: context.TODO(),
+		}
+		res, err := client.Default.Nodes.AddContainerNode(params)
+		require.Error(t, err) // Can't use EqualError because it returns different references each time.
+		require.Contains(t, err.Error(), "unknown error (status 400)")
+		require.Nil(t, res)
+	})
+}
+
+func TestRemoteNode(t *testing.T) {
+	t.Run("Basic", func(t *testing.T) {
+		params := &nodes.AddRemoteNodeParams{
+			Body: nodes.AddRemoteNodeBody{
+				NodeName: "Test Remote Node",
+			},
+			Context: context.TODO(),
+		}
+		res, err := client.Default.Nodes.AddRemoteNode(params)
+		require.NoError(t, err)
+		require.NotNil(t, res.Payload.Remote)
+		nodeID := res.Payload.Remote.NodeID
+		defer removeNodes(t, nodeID)
+
+		// Check node exists in DB.
+		getNodeRes, err := client.Default.Nodes.GetNode(&nodes.GetNodeParams{
+			Body:    nodes.GetNodeBody{NodeID: nodeID},
+			Context: context.TODO(),
+		})
+		require.NoError(t, err)
+		require.NotNil(t, getNodeRes.Payload.Remote)
+		require.Equal(t, getNodeRes.Payload.Remote.NodeID, nodeID)
+		require.Equal(t, getNodeRes.Payload.Remote.NodeName, params.Body.NodeName)
+
+		// Check duplicates.
+		res, err = client.Default.Nodes.AddRemoteNode(params)
+		require.Error(t, err) // Can't use EqualError because it returns different references each time.
+		require.Contains(t, err.Error(), "unknown error (status 409)")
+		require.Nil(t, res)
+
+		// Change node.
+		changeRes, err := client.Default.Nodes.ChangeRemoteNode(&nodes.ChangeRemoteNodeParams{
+			Body:    nodes.ChangeRemoteNodeBody{NodeID: nodeID, NodeName: "Changed Remote Node"},
+			Context: context.TODO(),
+		})
+		require.NoError(t, err)
+		require.NotNil(t, changeRes.Payload.Remote)
+		require.Equal(t, changeRes.Payload.Remote.NodeID, nodeID)
+		require.Equal(t, changeRes.Payload.Remote.NodeName, "Changed Remote Node")
+	})
+
+	t.Run("AddNameEmpty", func(t *testing.T) {
+		params := &nodes.AddRemoteNodeParams{
+			Body:    nodes.AddRemoteNodeBody{NodeName: ""},
+			Context: context.TODO(),
+		}
+		res, err := client.Default.Nodes.AddRemoteNode(params)
+		require.Error(t, err) // Can't use EqualError because it returns different references each time.
+		require.Contains(t, err.Error(), "unknown error (status 400)")
+		require.Nil(t, res)
+	})
+}
+
+func TestRemoteAmazonRDSNode(t *testing.T) {
+	t.Run("Basic", func(t *testing.T) {
+		params := &nodes.AddRemoteAmazonRDSNodeParams{
+			Body: nodes.AddRemoteAmazonRDSNodeBody{
+				NodeName: "Test RemoteAmazonRDS Node",
```

Comment:
This test will fail with two parallel runs – the second one will try to use the same name.

All unique identifiers should be generated. All testing strings (including generated ones) should start with "Test" or "test" or "test/" or something like that.

Probably, it worth to extract generators functions.